### PR TITLE
CELDEV-1076 - add celLazyLoader.mjs to default js and tinymce6 support

### PR DESCRIPTION
**Pull request by fpichler**
https://synjira.atlassian.net/browse/CELDEV-1076

### CODE REVIEW

File: src/main/webapp/templates/celAjax/picker/filePickerList.vm [hunk 1]
```diff
@@ -0,0 +1,44 @@
+$response.setStatus(200)
+#set($start = 0)
+#if(("$!request.start" != '') && ($util.parseInt("$!request.start") > 0))
+  #set($start = $util.parseInt("$!request.start"))
+#end
+#set($nb = 0)
+#if(("$!request.nb" != '') && ($util.parseInt("$!request.nb") > 0))
+  #set($nb = $util.parseInt("$!request.nb"))
+#end
```
Comment (msladek) [inline src/main/webapp/templates/celAjax/picker/filePickerList.vm]: `$util.parseInt` already defaults to 0 for empty/nonint strings:
```suggestion
#set($start = $mathtool.max(0, $util.parseInt("$!request.start")))
#set($nb = $mathtool.max(0, $util.parseInt("$!request.nb")))
```

File: src/main/webapp/templates/celAjax/picker/filePickerList.vm [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#elseif("$request.orderby" == 'ChangeDateDesc')
+  #set($comparator = 'AttachmentDescendingChangeDateComparator')
+#end
+#set($fileBaseDocRef = $services.reference.resolve("$!request.filebaseDocFN"))
+#set($fileBaseDoc = $services.modelAccess.getDocument($fileBaseDocRef))
+#set($onlyImages = ("$!request.onlyImages" == 'true'))
+#if("$!request.spaceImgs" != '')
+  #set($attachments = $xwiki.celementsweb.getAttachmentListForTagSortedSpace("$!request.spaceImgs", "$!request.tagList", $comparator, $onlyImages, $start, $nb))
+#else
+  #set($attachments = $xwiki.celementsweb.getAttachmentListForTagSorted($fileBaseDoc, "$!request.tagList", $comparator, $onlyImages, $start, $nb))
```
Comment (msladek) [inline src/main/webapp/templates/celAjax/picker/filePickerList.vm:20]: these methods are deprecated. are there no replacements in `$services.web/Utils`?

File: src/main/webapp/resources/celRes/ImageAndFilePicker/ImageAndFilePicker.css [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+.filebasePicker #attachments.celDropOverActive::after {
+    content: "drop files here";
+    position: absolute;
+    align-items: center;
+    display: block;
+    top: 50%;
+    left: 40%;
+    font-size: 50px;
+    font-weight: bold;
+    color: black;
```
Comment (msladek) [inline src/main/webapp/resources/celRes/ImageAndFilePicker/ImageAndFilePicker.css]:
```suggestion
  content: "drop files here";
  position: absolute;
  align-items: center;
  display: block;
  top: 50%;
  left: 40%;
  font-size: 50px;
  font-weight: bold;
  color: black;
```

File: src/main/webapp/resources/celJS/editResize.js [hunk 4]
```diff
@@ -108,6 +108,7 @@
   
     var boxSize = winHeight - bottomBorder;
     getScrollboxes().each(function(scrollbox){
+      console.log('debugging resize: scrollbox ', scrollbox);
```
Comment (msladek) [inline src/main/webapp/resources/celJS/editResize.js]:
```suggestion
      console.debug('debugging resize: scrollbox ', scrollbox);
```

File: src/main/webapp/resources/celDynJS/ImageAndFilePicker/ImageAndFilePicker.mjs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      imgDiv.setAttribute('data-att-src', attSrc);
+      imgDiv.setAttribute('data-filename', attElem.filename);
+      imgDiv.appendChild(imgContainerDiv);
+      if(options.duplicateCheck) {
+        attachEl.querySelectorAll('.imagePickerWrapper').forEach(function(imgWrapper) {
+          if(imgWrapper.querySelector('img').src.href == attElemUrl.href) {
+            imgWrapper.remove();
+          }
+        });
+      }
```
Comment (msladek) [inline src/main/webapp/resources/celDynJS/ImageAndFilePicker/ImageAndFilePicker.mjs]: some code style suggestions:
```suggestion
      if (options.duplicateCheck) {
        [...attachEl.querySelectorAll('.imagePickerWrapper')]
          .filter(imgWrapper => imgWrapper.querySelector('img').src.href == attElemUrl.href)
          .forEach(imgWrapper => imgWrapper.remove());
      }
```

File: src/main/webapp/resources/celDynJS/ImageAndFilePicker/ImageAndFilePicker.mjs [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    formData.append('start', this.startPos);
+    formData.append('nb', this.stepNumber);
+    formData.append('tagList', this.tag);
+    formData.append('onlyImages', this.onlyImages)
+    formData.append('filebaseDocFN', this.filebaseFN)
+    fetch('/ajax/picker/filePickerList', {
+      method : 'POST',
+      body : formData
+    }).then(resp => resp.json()
+    ).then(data => {
```
Comment (msladek) [inline src/main/webapp/resources/celDynJS/ImageAndFilePicker/ImageAndFilePicker.mjs]: why no async/await?

File: src/main/webapp/resources/celDynJS/ImageAndFilePicker/ImageAndFilePicker.mjs [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    const formData = new FormData();
+    formData.append('xpage', 'celements_ajax');
+    formData.append('ajax_mode', 'imagePickerList');
+    formData.append('images', '1');
+    formData.append('start', this.startPos);
+    formData.append('nb', this.stepNumber);
+    formData.append('tagList', this.tag);
+    formData.append('onlyImages', this.onlyImages)
+    formData.append('filebaseDocFN', this.filebaseFN)
+    fetch('/ajax/picker/filePickerList', {
```
Comment (msladek) [inline src/main/webapp/resources/celDynJS/ImageAndFilePicker/ImageAndFilePicker.mjs]: return the promise, a caller might want to await it.

File: src/main/webapp/resources/celDynJS/ImageAndFilePicker/ImageAndFilePicker.mjs [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+        'clickHandler' : (event) => {
+          const imageDiv = event.currentTarget;
+          this.callback(imageDiv.dataset.attSrc, { alt : imageDiv.dataset.filename });
+          this.pickerOverlay.close();
+        }
+      });
+      this.dropHandler.registerHandler(attachEl);
+      this.pickerOverlay.updateContent([attachEl]);
+    });
+  }
```
Comment (msladek) [inline src/main/webapp/resources/celDynJS/ImageAndFilePicker/ImageAndFilePicker.mjs:86]: the methods `renderAttachmentList` and `updateAttachmentList` are fairly long and hard to optically parse. can they be further split?
maybe creating the HTML with [template string interpolation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#string_interpolation) might also improve readability.

**Review by msladek — CHANGES_REQUESTED** [review]

File: src/main/webapp/resources/celDynJS/ImageAndFilePicker/ImageAndFilePicker.mjs (hunk 8 above)
Comment (fpichler) [reply src/main/webapp/resources/celDynJS/ImageAndFilePicker/ImageAndFilePicker.mjs:86]: see CELDEV-1078

File: src/main/webapp/templates/celAjax/picker/filePickerList.vm (hunk 2 above)
Comment (fpichler) [reply src/main/webapp/templates/celAjax/picker/filePickerList.vm:20]: unfortunately not.

Comment (msladek) [reply src/main/webapp/templates/celAjax/picker/filePickerList.vm:20]: ?

**Review by msladek — APPROVED** [review]

File: src/main/webapp/templates/celAjax/picker/filePickerList.vm (hunk 2 above)
Comment (fpichler) [reply src/main/webapp/templates/celAjax/picker/filePickerList.vm:20]: `CelementsWebPluginApi.getAttachmentListForTagSorted` and `CelementsWebPluginApi.getAttachmentListForTagSortedSpace` are the only entry point for scripts. In the deprecation note `WebUtilsService.getAttachmentListSorted` is suggested as a replacement. First WebUtilsService is not accessible from scripts, second it `getAttachmentListSorted` is missing the tag filtering.

**Pull request merged**